### ISSUE TITLE
Failure on windows XP and 2003 while working on later version

### ISSUE THREAD

**Issue by stefb69** [chocolatey/puppet-chocolatey#41]
I've come across a problem that i can't seem to fix easily

I'm using the chocolatey provider to deploy applications on a few windows machines, and everything works very well on windows 7 and 8.1 and 2008 R2, but when run on XP 32bit or 2003 32 bit i get an  error `undefined method 'each' for nil:NilClass` : 

* Powershell 2.0 minimum is installed
* Using chocolatey on the command line works fine
* All systems are fr-FR versions of windows
* Puppet master is at version 3.7.3 and clients are running Puppet agent 3.3.1 on XP and 3.7.3 on win 2003, 7, 8.1 etc....
* I'm using `fix_latest` branch, but the i've comme across the same issue with the master branch

The this the class used to install/update packages 
```puppet
class winbrowser {
  $pkg = [ 'flashplayerplugin', 'firefox' ]
  
  package { $pkg:
    ensure   => 'latest',
    provider => 'chocolatey',
    source   => 'https://chocolatey.org/api/v2/'
  }
}
```

```log
Could not prefetch package provider 'chocolatey': undefined method 'each' for nil:NilClass
//Stage[main]/Winbrowser/Package[firefox]) Could not evaluate: undefined method 'each' for nil:NilClass
```

### Workaround
tl;dr: Workaround noted at https://github.com/chocolatey/puppet-chocolatey/issues/41#issuecomment-63387953

**Comment by karaaie** [comment]
I seem to be getting the same problem
Agent is running Windows Server 2012 R2, 
Puppet agent version is 3.7.4
Foreman gives this error message
```
 	Could not prefetch package provider 'chocolatey': undefined method `each' for nil:NilClass
```
However the puppet run works flawlessly when I run it from the command line in powershell.
And I'm not running PE, I'm using the regular Puppet Server. I tried changing 
``` ruby -rubygems "%~dp0daemon.rb" %*``` in C:\Program Files (x86)\Puppet Labs\Puppet\service\daemon.bat to rubyw but it didnt seem to help. Or was I supposed to restart the puppet service first?

**Comment by ferventcoder** [comment]
Closing as an MCO error that was resolved.

**Comment by ferventcoder** [comment]
Sorry, that was for the second issue. Not sure if the original issue was closed or not.

**Comment by ferventcoder** [comment]
I believe this is fixed, but if not, please feel free to comment. Anyone else coming across this - if you are using Windows 2003, please look to another directory besides the default. It is pretty deep down the folder structure and you will start running into path too long issues.
